modules/phase_field/src/ics/SpecifiedSmoothSuperellipsoidIC.C

```diff
@@ -0,0 +1,73 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "SpecifiedSmoothSuperellipsoidIC.h"
+#include "MooseRandom.h"
+
+template<>
+InputParameters validParams<SpecifiedSmoothSuperellipsoidIC>()
+{
+  InputParameters params = validParams<SmoothSuperellipsoidBaseIC>();
+  params.addClassDescription("Multiple smooth superellipsoids with manually specified center points; semiaxes a,b,c; and exponents n");
+  params.addRequiredParam<std::vector<Real> >("x_positions", "The x-coordinate for each superellipsoid center");
+  params.addRequiredParam<std::vector<Real> >("y_positions", "The y-coordinate for each superellipsoid center");
+  params.addRequiredParam<std::vector<Real> >("z_positions", "The z-coordinate for each superellipsoid center");
```

Comment:
Yeah, what he has is fine. I'm just saying that if these really are points, you can read them in as Points